Add Mul postprocessor for Vector

jhaayushkumar opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->

Related to gh-26121 and gh-26623.

#### Brief description of what is fixed or changed

When [simplify()](cci:1://file:///Users/ayushkumarjha/Desktop/sympy/sympy/vector/tests/test_vector.py:193:0-225:60) traverses expression trees using `bottom_up()`, it reconstructs nodes via `.func(*args)`. This caused [VectorMul](cci:2://file:///Users/ayushkumarjha/Desktop/sympy/sympy/vector/vector.py:520:0-539:35) objects to become [Mul](cci:2://file:///Users/ayushkumarjha/Desktop/sympy/sympy/vector/vector.py:520:0-539:35) objects, which then failed type checks in `CoordSys3D.__new__()` with `TypeError: location should be a Vector`.

This PR adds a postprocessor for [Mul](cci:2://file:///Users/ayushkumarjha/Desktop/sympy/sympy/vector/vector.py:520:0-539:35) (similar to the existing [Add](cci:2://file:///Users/ayushkumarjha/Desktop/sympy/sympy/vector/vector.py:498:0-517:27) postprocessor from gh-26623) to ensure [Mul](cci:2://file:///Users/ayushkumarjha/Desktop/sympy/sympy/vector/vector.py:520:0-539:35) operations on vectors return [VectorMul](cci:2://file:///Users/ayushkumarjha/Desktop/sympy/sympy/vector/vector.py:520:0-539:35) objects.



#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->

* vector
  * Fixed TypeError when simplifying vector expressions with spherical coordinates and `locate_new()`. VectorMul objects are now preserved during simplification.

<!-- END RELEASE NOTES -->

jhaayushkumar commented:
In the previous approach, using `VectorMul(*vectors)` only passed vector arguments,causing scalar coefficients `(like the 2 in Mul(2, C.i)` to be dropped. I've updated it to `VectorMul(*nonvectors, *vectors)`, which now correctly passes both scalars and vectors to preserve the complete expression.

Additionally, I added a check to only process "simple" vectors (BaseVector, VectorMul, VectorAdd, VectorZero) and exclude unevaluated operations like Cross/Dot. This prevents infinite recursion that would occur when negating Cross objects.

Did the approach is fine or need some more refining.

oscarbenjamin commented on sympy/vector/vector.py:
```diff
@@ -418,19 +418,31 @@ def _div_helper(one, other):
         if vec_class == VectorAdd:
             return VectorAdd(*vectors).doit(deep=False)
+        if vec_class == VectorMul:
+            return VectorMul(*nonvectors, *vectors).doit(deep=False)
```
I don't see why this needs to separate vectors and nonvectors at all if they are all going to be passed to VectorMul.

Think more carefully about how to write this code because I think it can be much simpler than this.

jhaayushkumar committed:
Fix vector multiplication postprocessor for Mul expressions

Add Mul postprocessor to handle scalar-vector multiplication correctly.
This fixes TypeError during simplification of vector expressions with
spherical coordinates.

Fixes #28727

jhaayushkumar committed:
Refactor vector multiplication test by removing comments

Remove comments and assertions related to vector multiplication regression test.

jhaayushkumar committed:
Update test_vector.py
